### PR TITLE
Add CRC16 CCITT False Implementation

### PR DESCRIPTION
**Pull request by kuqin12**
# Preface

Please ensure you have read the [contribution docs](https://github.com/microsoft/mu/blob/master/CONTRIBUTING.md) prior
to submitting the pull request. In particular,
[pull request guidelines](https://github.com/microsoft/mu/blob/master/CONTRIBUTING.md#pull-request-best-practices).

## Description

This change is added to incorporate basic implementation for CRC16-CCITT-FALSE algorithm.

This function is useful for providing CRC16 value used in other data structures that requires CRC16 value that complies with JEDEC SPD requirements, i.e. BDAT table.

The lookup table is inherited from `https://crccalc.com/` and the result values are also compared against this site.

For each item, place an "x" in between `[` and `]` if true. Example: `[x]`.
_(you can also check items in the GitHub UI)_

- [x] Impacts functionality?
  - **Functionality** - Does the change ultimately impact how firmware functions?
  - Examples: Add a new library, publish a new PPI, update an algorithm, ...
- [ ] Impacts security?
  - **Security** - Does the change have a direct security impact on an application,
    flow, or firmware?
  - Examples: Crypto algorithm change, buffer overflow fix, parameter
    validation improvement, ...
- [ ] Breaking change?
  - **Breaking change** - Will anyone consuming this change experience a break
    in build or boot behavior?
  - Examples: Add a new library class, move a module to a different repo, call
    a function in a new library class in a pre-existing module, ...
- [ ] Includes tests?
  - **Tests** - Does the change include any explicit test code?
  - Examples: Unit tests, integration tests, robot tests, ...
- [ ] Includes documentation?
  - **Documentation** - Does the change contain explicit documentation additions
    outside direct code modifications (and comments)?
  - Examples: Update readme file, add feature readme file, link to documentation
    on an a separate Web page, ...

## How This Was Tested

The function output is compared and matches with the results of 3rd party online CRC calculators.

## Integration Instructions

N/A

### CODE REVIEW

File: MdePkg/Library/BaseLib/CheckSum.c [hunk 1]
```diff
@@ -762,3 +762,71 @@ CalculateCrc32c (
...
+CalculateCrc16CcittF (
+  IN CONST VOID  *Buffer,
+  IN UINTN       Length,
+  IN UINT16      InitialValue
+  )
+{
+  CONST UINT8  *Buf;
+  UINT16       Crc;
+
+  Buf = Buffer;
```
Comment (apop5) [inline MdePkg/Library/BaseLib/CheckSum.c:825]: Buffer needs validated before it's used

**Review by apop5 — APPROVED** [review]

File: MdePkg/Library/BaseLib/CheckSum.c (hunk 1 above)
Comment (kuqin12) [reply MdePkg/Library/BaseLib/CheckSum.c:825]: I thought about adding it, but any idea on what i should return if validation fails?

Comment (apop5) [reply MdePkg/Library/BaseLib/CheckSum.c:825]: CalculateCrc32 (in the same file) does the following:

  ASSERT (Buffer != NULL);
  ASSERT (Length <= (MAX_ADDRESS - ((UINTN)Buffer) + 1));
  
  I think we could just do the same here, and then return 0 if it is NULL.

Comment (kuqin12) [reply MdePkg/Library/BaseLib/CheckSum.c:825]: Will update shortly, thanks!

Comment (kuqin12) [reply MdePkg/Library/BaseLib/CheckSum.c:825]: Updated~

**Review by cfernald — APPROVED** [review]

**Pull request merged**